### PR TITLE
Update Elasticsearch docs for Heroku

### PR DESCRIPTION
**Pull request by mraible**

### CODE REVIEW

File: pages/using_elasticsearch.md [hunk 1]
```diff
@@ -45,3 +52,30 @@ In production, JHipster expects an external Elasticsearch instance. By default,
 ### Using on Heroku
 
 On Heroku, the [Bonsai Elasticsearch](https://elements.heroku.com/addons/bonsai) is configured as an add-on. JHipster is automatically configured to talk to it.
```
Comment (atomfrede) [inline pages/using_elasticsearch.md:54]: Should we remove this? Or rephrase it to something like "In Heroku, JHipster used to configure Bonsai. Unfortunately..."

File: pages/using_elasticsearch.md [hunk 2]
```diff
@@ -45,3 +52,30 @@ In production, JHipster expects an external Elasticsearch instance. By default,
...
+
+Download your credentials from the next screen and click **Continue**. Then, select **Manage this deployment** from the menu and copy the Elasticsearch endpoint. 
+
+Set the credentials and endpoint URL as a new `ELASTIC_URL` environment variable on Heroku.
+
+```shell
+heroku config:set ELASTIC_URL=https://elastic:<password>@<endpoint-url>
+```
+
+Then, modify `heroku.gradle` to remove the workaround for Bansai (that no longer works) and update `application-heroku.yml` to use `ELASTIC_URL`:
```
Comment (atomfrede) [inline pages/using_elasticsearch.md:72]:
```suggestion
Then, modify `heroku.gradle` to remove the workaround for Bonsai (that no longer works) and update `application-heroku.yml` to use `ELASTIC_URL`:
```

I will remove the workaround from both gradle and maven in a separate PR.

File: pages/using_elasticsearch.md [hunk 3]
```diff
@@ -45,3 +52,30 @@ In production, JHipster expects an external Elasticsearch instance. By default,
 ### Using on Heroku
 
 On Heroku, the [Bonsai Elasticsearch](https://elements.heroku.com/addons/bonsai) is configured as an add-on. JHipster is automatically configured to talk to it. 
+
+Unfortunately, as of JHipster 7.9.3, Elasticsearch [doesn't work out of the box with Heroku](https://github.com/jhipster/generator-jhipster/issues/20315). To solve this, you can create a Docker image with Elasticsearch and deploy it to somewhere that can run it, or you can use Elastic Cloud.
```
Comment (atomfrede) [inline pages/using_elasticsearch.md:56]: Maybe one sentence why we don't describe the elastic heroku add-on? (because of pricing, no free trial?)

**Review by atomfrede — COMMENTED** [review]
Just minor things

File: pages/using_elasticsearch.md (hunk 1 above)
Comment (mraible) [reply pages/using_elasticsearch.md:54]: Currently, it's accurate. I think we could change it, but we should change the code first.

File: pages/using_elasticsearch.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ redirect_from:
   - /using_elasticsearch.html
 sitemap:
     priority: 0.7
-    lastmod: 2017-04-16T00:00:00-00:00
+    lastmod: 2022-11-28T00:00:00-00:00
```
Comment (mraible) [inline pages/using_elasticsearch.md:9]:
```suggestion
    lastmod: 2022-12-13T00:00:00-00:00
```

File: pages/using_elasticsearch.md (hunk 3 above)
Comment (mraible) [inline pages/using_elasticsearch.md:56]:
```suggestion
Unfortunately, as of JHipster 7.9.3, Elasticsearch [doesn't work out of the box with Heroku](https://github.com/jhipster/generator-jhipster/issues/20315). To solve this, you can create a Docker image with Elasticsearch and deploy it to somewhere that can run it, or you can use Elastic Cloud. We don't automatically configure the [Elasticsearch Add-on](https://elements.heroku.com/addons/foundelasticsearch) because cheapest plan is 67 USD/month.
```

**Comment by mraible** [comment]
@atomfrede Can you please review again?

File: pages/using_elasticsearch.md [hunk 5]
```diff
@@ -17,26 +17,33 @@ This option has some limitations:
...
+    *   This also means if your database is changed outside of your application, your search indexes will be out-of-sync. The [Elasticsearch Reindexer](https://www.jhipster.tech/modules/marketplace/#/details/generator-jhipster-elasticsearch-reindexer) JHipster module can help in these situations.
 
 When the Elasticsearch option is selected:
 
-*   Spring Data Elasticsearch is used, with the help of [Spring Data Jest](https://github.com/VanRoy/spring-data-jest). Spring Data Jest which allows communication with Elasticsearch's REST API. It disables Spring Boot's autoconfiguration and uses its own instead.
-*   The "repository" package has new subpackage, called "search", that holds all Elasticsearch repositories.
+*   Spring Data Elasticsearch is used with Spring Boot's autoconfiguration.
+*   The "repository" package has a new subpackage, called "search", that holds all Elasticsearch repositories.
 *   The "User" entity gets indexed in Elasticsearch, and you can query is using the `/api/_search/users/:query` REST endpoint.
```
Comment (atomfrede) [inline pages/using_elasticsearch.md:26]:
```suggestion
*   The "User" entity gets indexed in Elasticsearch, and you can query it using the `/api/_search/users/:query` REST endpoint.
```

**Review by atomfrede — APPROVED** [review]
Reads good to me. Just a minor typo (which seems to be rather old)
